robot/robot.go

```diff
@@ -57,24 +58,49 @@ func (r *Robot) Move(steps int) error {
 	if r.EnvMap.Map == "" {
 		return fmt.Errorf("The robot can't move because no environmental map is given")
 	}
+	err := r.tryMoveToPosition(steps)
+	if err != nil {
+		return err
+	}
+	distance := Centimeters(float64(steps) * float64(r.EnvMap.SquareSize))
+	r.MoveModule.Move(distance, r.Tachometer)
+	return nil
+}
+
+func (r *Robot) tryMoveToPosition(steps int) error {
 	newPosition := r.Position
+	absSteps := getAbsoluteSteps(steps)
+	singleStep := getMovingDirection(steps)
+	for i := 0; i < absSteps; i++ {
+		newPosition = r.moveSingleStepFrom(newPosition, singleStep)
+		if r.EnvMap.isObstacleAt(newPosition) {
+			return fmt.Errorf("The robot can't move %v steps because there are obstacles in the way.Here's the current map: %v", steps, mapWithRobot(r))
+		}
+	}
+	r.Position = newPosition
+	return nil
+}
+
+func (r *Robot) moveSingleStepFrom(p Position, step int) Position {
```

Comment:
I'm still not quite happy with the step parameter, but I see no other way around to keep the robot able to move backwards. Any suggestions?